### PR TITLE
Update build script to detect when translation is needed

### PR DESCRIPTION
**Pull request by mgramigna**
Build script will now use the modified Time of the file descriptor for both the ELM and CQL to determine when translation is needed (i.e. when CQL is modified more recently than existing ELM, _if_ existing ELM is present in the build directory).

To test, I recommend just starting a docker container and using the `yarn translate` command for various cases described below:

```
docker run -p 8080:8080 -d cqframework/cql-translation-service:latest
```

1. After a fresh pull, `yarn translate` should skip translation since there are no changes
2. Modify mcode.cql. `yarn translate` should translate and write the elm
3. Do nothing, just run `yarn translate` again, and it should skip translation
4. Create a brand new cql file, `yarn translate` should pick it up and write the elm
5. Do nothing, just run `yarn translate` again, and it should skip translation for both mcode and the new CQL file

There could be some cases other than the process outlined there, but you get this gist.

### CODE REVIEW

**Review by Dtphelan1 — APPROVED** [review]
This looks great to me. Code is readable, and works as expected on my local machine.

**Review by jafeltra — COMMENTED** [review]
This looks good to me! The one thing that has confused me is that I switched to this branch and ran `yarn translate` on everything that's checked in, and I got a new ELM file written for `mCODE.cql`. That happened more than once, but now I can't get it to happen again, so I think I'm willing to believe it was something weird on my end.

The only other thing I thought of if we want to remove an ELM file for a CQL file that was removed. But it doesn't look like we do that on `master` now anyway. Do you agree that's separate (if even desired at all)?

**Comment by mgramigna** [comment]
For JA1, my best guess is that the current state of the code probably just has CQL changes without them represented in the corresponding ELM so it's probably a special case that can be ignored /shrug

For JA2, I feel like it's dealers choice for this. I think it's logical/reasonable behavior for the build directory to _only_ contain translated ELM for what currently exists in src. Do we forsee any consequences with doing just an `rm -rf <build_path>` before doing translation? Now that I just typed that out, the build path is eventually going to be something that's configurable by the user of the testing harness, so it's possible they would want to put the ELM in a non-empty directory that contains things other than just ELM.... right?

TL;DR I might've just convinced myself that removing anything in the build path directory might not be desirable cause we don't quite know what else a user might have manually put in there.

**Comment by jafeltra** [comment]
Hm yeah I agree that we probably don't want to do JA2, because if we're also wiping out the build directory, doesn't that imply that we'd always need to rebuild the ELM, which defeats the purpose of this task. So I'm good with this as it is!

**Review by jafeltra — APPROVED** [review]
